### DISCUSSION TITLE
Scope of support for networks consisting of multiple node versions

### DISCUSSION THREAD

**Discussion by YoshihitoAso** [ConsenSys/quorum#1420] [Q&A] [answered]
How well do you test a network with multiple mixed node versions with each upgrade?

Our testing results show that with the Istanbul consensus adopted, when upgrading a validator node from v21.1 to v21.4 or higher, all nodes need to be upgraded, including the non-validator node.

See: https://github.com/BoostryJP/ibet-Network/issues/96 (results are in Japanese only)

The v21.4 release notes do not describe this point. I assume it is not included in the test scope of the upgrade.

**Comment by antonydenyer** [comment, marked as answer]
We only test the last two major versions. We also have some tests between major milestone (e.g upgrades in the eth protocol).

However, this issue was raised https://github.com/ConsenSys/quorum/issues/1405 and should be fixed since https://github.com/ConsenSys/quorum/releases/tag/v22.4.3.
